Sources/Data/Model/Event.swift

```diff
@@ -35,3 +35,64 @@ public final class Event : NSManagedObject {
         self.timestamp = Date()
     }
 }
+
+extension Event {
+    /// Register a callback to be fired whenever Events are inserted.
+    ///
+    /// Note that this returns an opaque chit object that you must retain until you no longer wish the callback to be fired.
+    public static func observeNewEvents(
+        managedObjectContext: NSManagedObjectContext,
+        observerCallback: @escaping ([Event]) -> Void
+    ) -> NSObjectProtocol {
+        return NotificationCenter.default.addObserver(forName: .NSManagedObjectContextObjectsDidChange, object: managedObjectContext, queue: nil) { (iosNotification) in
+            guard let insertedObjects = iosNotification.userInfo?[NSInsertedObjectsKey] as? Set<NSObject> else {
+                return
+            }
+            
+            let insertedEvents = insertedObjects
+                .compactMap({ $0 as? Event })
+            
+            if !insertedEvents.isEmpty {
+                observerCallback(insertedEvents)
+            }
+        }
+    }
+    
+    public func attemptInsert() {
+        guard let managedObjectContext = self.managedObjectContext else {
+            assertionFailure("Not associated with a managed object context, cannot insert.")
+            return
+        }
+        
+        managedObjectContext.perform {
```

Comment:
Should this not be using your new `saveOrRollback` method defined in your extension?